logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt

```diff
@@ -36,6 +38,21 @@ class MLSConversationDataSource(
     private val conversationMapper: ConversationMapper = MapperProvider.conversationMapper()
 ): MLSConversationRepository {
 
+    override suspend fun messageFromMLSMessage(messageEvent: Event.Conversation.NewMLSMessage): Either<CoreFailure, ByteArray?> = suspending {
+        mlsClientProvider.getMLSClient()
+            .flatMap { mlsClient ->
+                wrapStorageRequest {
+                    conversationDAO.getConversationByQualifiedID(idMapper.toDaoModel(messageEvent.conversationId)).first()
+                }.flatMap { conversation ->
+                    if (conversation.protocolInfo is ConversationEntity.ProtocolInfo.MLS) {
+                        Either.Right(mlsClient.decryptMessage((conversation.protocolInfo as ConversationEntity.ProtocolInfo.MLS).groupId, messageEvent.content.decodeBase64Bytes()))
```

Comment:
I wonder if MLS throws exceptions if enc/decryption is kaput.

Reply:
It does, I'll add a TODO to create something similar like we do for proteus.